### PR TITLE
Support supplementary rulesets in battles

### PR DESCRIPTION
**Pull request by sirDonovan**

### CODE REVIEW

**Comment by Slayer95** [comment]
Those should be Custom Games, shouldn't they?
Also: https://github.com/Zarel/Pokemon-Showdown/issues/3201

**Comment by sirDonovan** [comment]
I don't think they need to be. It's easier to add a few changes to standard formats than to add everything manually to Custom Game.

In regards to that PR, it doesn't seem like we've fully decided on a solution. For now we can just disable logging for battles with altered rulesets?

**Comment by Zarel** [comment]
We already only log ladder games, which these are exempt from.

**Comment by Zarel** [comment]
Right now custom rulesets are only supported for tournaments; I've been meaning to support them in challenges but I haven't gotten to that yet.

**Comment by Zarel** [comment]
I'm not sure if this is the best way to implement custom rulesets, but it probably isn't the worst.

**Comment by TheImmortal** [comment]
Would be cool if we could set a custom name for Tournaments in the case of these special rules. I'd rather not do an OU tour and announce a couple of times that it is an Inverse tour.

**Comment by sirDonovan** [comment]
@TheImmortal I'll look into that next.

**Comment by TheImmortal** [comment]
@Zarel is this fine

File: battle-engine.js [hunk 1]
```diff
@@ -4676,7 +4676,7 @@ exports.BattleSide = BattleSide;
 exports.Battle = Battle;
 
 let battleProtoCache = new Map();
-exports.construct = function (format, rated, send) {
+exports.construct = function (format, rated, supplementaryRuleset, send) {
```
Comment (Zarel) [inline battle-engine.js]: A `send` function is so passé though...

I think maybe the format should be constructed outside of this function, so this function's signature stays unchanged.

**Comment by sirDonovan** [comment]
How about this?

**Comment by Zarel** [comment]
Custom formats definitely need to be redone at some point, but I'll assume maintenance burden on this.

**Pull request merged**